src/main/java/com/teamwizardry/refraction/client/core/RenderLaserUtil.java

```diff
@@ -40,64 +40,41 @@ public static void finishRenderingLasers() {
 		Tessellator.getInstance().draw();
 	}
 
-	public static void renderLaser(Color color, Vec3d start, Vec3d end) {
+	public static void renderLaser(Color color, Vec3d start, Vec3d end, BufferBuilder bb) {
 		if (ConfigValues.USE_FLAT_BEAM_TEXTURE) Minecraft.getMinecraft().getTextureManager().bindTexture(textLaserFlat);
 		else Minecraft.getMinecraft().getTextureManager().bindTexture(textLaser);
 
 		int alpha = Math.max(50, color.getAlpha());
 		int addColorMin = 30 * alpha / 255;
 		color = new Color(Math.max(addColorMin, color.getRed()), Math.max(addColorMin, color.getGreen()), Math.max(addColorMin, color.getBlue()), alpha);
 
-		GlStateManager.disableCull();
-		if (ConfigValues.ADDITIVE_BLENDING) {
-			GlStateManager.enableBlend();
-			GlStateManager.blendFunc(GL_SRC_ALPHA, GL_ONE);
-		}
-
-		Vec3d playerEyes = Minecraft.getMinecraft().player.getPositionEyes(ClientTickHandler.getPartialTicks());
-		Vec3d normal = (end.subtract(start)).crossProduct(playerEyes.subtract(start)).normalize(); //(b.subtract(a)).crossProduct(c.subtract(a));
-		if (normal.y < 0)
-			normal = normal.scale(-1);
+		// Vec3d playerEyes = Minecraft.getMinecraft().player.getPositionEyes(ClientTickHandler.getPartialTicks());
+		// Vec3d normal = (end.subtract(start)).crossProduct(playerEyes.subtract(start)).normalize(); //(b.subtract(a)).crossProduct(c.subtract(a));
+		// if (normal.y < 0)
+		// 	   normal = normal.scale(-1);
 
 		//Vec3d d = normal.scale((0.25 * color.getAlpha() / 255f) / 2.);
-		Vec3d d = new Vec3d(0, (0.25 * color.getAlpha() / 255f) / 2.0, 0);
-		Vec3d d2 = new Vec3d((0.25 * color.getAlpha() / 255f) / 2.0, 0, 0);
-		Vec3d d3 = new Vec3d(0, 0, (0.25 * color.getAlpha() / 255f) / 2.0);
+		Vec3d d = new Vec3d(0, (0.25 * color.getAlpha() / 255D) / 2.0, 0);
+		Vec3d d2 = new Vec3d((0.25 * color.getAlpha() / 255D) / 2.0, 0, 0);
+		Vec3d d3 = new Vec3d(0, 0, (0.25 * color.getAlpha() / 255D) / 2.0);
```

Comment:
what is your issue with floats? :D

Reply:
I think it will avoid unnecessary casting?

Reply:
fair enough.